test/extensions/clusters/aggregate/cluster_integration_test.cc

```diff
@@ -303,5 +357,564 @@ TEST_P(AggregateIntegrationTest, PreviousPrioritiesRetryPredicate) {
+
+  // send the third request directly to cluster1
+  auto cluster1_response1 = codec_client_->makeHeaderOnlyRequest(Http::TestRequestHeaderMapImpl{
+      {":method", "GET"}, {":path", "/cluster1"}, {":scheme", "http"}, {":authority", "host"}});
+
+  // wait for the third request to arrive at cluster1
+  waitForNextUpstreamRequest(FirstUpstreamIndex);
+  // respond to the third request with a 503 to trigger a retry
+  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "503"}}, true);
+
+  // wait for the third request retry to arrive at cluster1
+  waitForNextUpstreamRequest(FirstUpstreamIndex);
+  auto& third_request_retry = *upstream_request_;
+
+  // the aggregate_cluster the circuit breaker remains open
+  test_server_->waitForGaugeEq("cluster.aggregate_cluster.circuit_breakers.default.rq_retry_open",
+                               1);
+  test_server_->waitForGaugeEq(
+      "cluster.aggregate_cluster.circuit_breakers.default.remaining_retries", 0);
+  test_server_->waitForCounterEq("cluster.aggregate_cluster.upstream_rq_retry_overflow", 1);
+  // the cluster1 circuit breaker opens
+  test_server_->waitForGaugeEq("cluster.cluster_1.circuit_breakers.default.rq_retry_open", 1);
+  test_server_->waitForGaugeEq("cluster.cluster_1.circuit_breakers.default.remaining_retries", 0);
+  test_server_->waitForCounterEq("cluster.cluster_1.upstream_rq_retry_overflow", 0);
+
+  // send the fourth request directly to cluster1
+  auto cluster1_response2 = codec_client_->makeHeaderOnlyRequest(Http::TestRequestHeaderMapImpl{
+      {":method", "GET"}, {":path", "/cluster1"}, {":scheme", "http"}, {":authority", "host"}});
+
+  // wait for the fourth request to arrive at cluster1
+  waitForNextUpstreamRequest(FirstUpstreamIndex);
+  // respond to the fourth request with a 503 to trigger a retry
+  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "503"}}, true);
+
+  // the aggregate_cluster the circuit breaker remains open
+  test_server_->waitForGaugeEq("cluster.aggregate_cluster.circuit_breakers.default.rq_retry_open",
+                               1);
+  test_server_->waitForGaugeEq(
+      "cluster.aggregate_cluster.circuit_breakers.default.remaining_retries", 0);
+  test_server_->waitForCounterEq("cluster.aggregate_cluster.upstream_rq_retry_overflow", 1);
+  // the cluster1 circuit breaker remains open and overflows
+  test_server_->waitForGaugeEq("cluster.cluster_1.circuit_breakers.default.rq_retry_open", 1);
+  test_server_->waitForGaugeEq("cluster.cluster_1.circuit_breakers.default.remaining_retries", 0);
+  test_server_->waitForCounterEq("cluster.cluster_1.upstream_rq_retry_overflow", 1);
+
+  // respond to the third request to cluster1
+  third_request_retry.encodeHeaders(default_response_headers_, true);
+  ASSERT_TRUE(cluster1_response1->waitForEndStream());
+  EXPECT_EQ("200", cluster1_response1->headers().getStatusValue());
+  // respond to the fourth request to cluster1
```

Comment:
This isn't a response I think? It's just the request failing, so it's more like "handle the response" instead of responding. <3